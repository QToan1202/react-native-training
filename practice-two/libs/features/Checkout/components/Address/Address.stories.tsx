import { ComponentMeta, ComponentStory } from '@storybook/react-native'
import { action } from '@storybook/addon-actions'

import { Address } from '../index'

export default {
  title: 'components/Address',
  component: Address,
} as ComponentMeta<typeof Address>

const Template: ComponentStory<typeof Address> = (args) => <Address {...args} />

export const Default = Template.bind({})
Default.args = {
  name: 'Deliver to Tradly Team, 75119',
  streetAddress: 'Kualalumpur, Malaysia',
  onPress: action('change address'),
}
