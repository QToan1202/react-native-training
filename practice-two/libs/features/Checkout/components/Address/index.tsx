import { memo } from 'react'
import isEqual from 'react-fast-compare'
import { XStack, XStackProps, YStack, getTokens } from 'tamagui'

import Button from '@components/Button'
import Paragraph from '@components/Paragraph'

export type AddressProps = {
  name: string
  streetAddress: string
  onPress: () => void
} & XStackProps

const Address = ({ name, streetAddress, onPress, ...rest }: AddressProps) => {
  return (
    <XStack
      justifyContent="space-between"
      alignItems="center"
      space={4}
      paddingVertical={15}
      paddingHorizontal={20}
      backgroundColor="$color.white"
      {...rest}
    >
      <YStack space={4}>
        <Paragraph fontWeight="$2" color="$gray_50" lineHeight={20} content={name} />
        <Paragraph
          fontWeight="$2"
          color="$gray_50"
          lineHeight={20}
          opacity={0.7}
          content={streetAddress}
        />
      </YStack>
      <Button
        paddingVertical={4}
        paddingHorizontal={getTokens().space[5].val - 1}
        onPress={onPress}
        fontSize="$1"
        title="Change"
      />
    </XStack>
  )
}

export default memo(Address, isEqual)
