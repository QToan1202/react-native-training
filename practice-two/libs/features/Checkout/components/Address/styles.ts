import { StyleSheet } from 'react-native'

import { COLORS, FONT_FAMILY } from '@practice-two/shared'

const styles = StyleSheet.create({
  container: {
    paddingVertical: 15,
    paddingHorizontal: 20,
    backgroundColor: COLORS.WHITE,
  },
  btn: {
    paddingHorizontal: 23,
    paddingVertical: 4,
  },
  btnTitle: {
    fontSize: 12,
  },
  text: {
    fontFamily: FONT_FAMILY.MONTSERRAT[500],
    color: COLORS.GRAY_50,
  },
  address: {
    opacity: 0.7,
  },
  spacing: {
    rowGap: 4,
  },
})

export default styles
