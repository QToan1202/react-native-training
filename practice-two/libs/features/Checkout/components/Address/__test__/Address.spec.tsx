import { fireEvent, render, screen } from '@testing-library/react-native'

import Address from '../index'

const mockOnPress = jest.fn()

describe('Testing Address component', () => {
  beforeEach(() => {
    render(
      <Address
        name="Deliver to Tradly Team, 75119"
        streetAddress="Kualalumpur, Malaysia"
        onPress={mockOnPress}
        testID="address-info"
      />
    )
  })

  it('Render component successfully', () => {
    const component = render(
      <Address
        name="Deliver to Tradly Team, 75119"
        streetAddress="Kualalumpur, Malaysia"
        onPress={mockOnPress}
      />
    ).toJSON()

    expect(component).toMatchSnapshot()
  })

  it('Component must have the right content', () => {
    const heading = screen.getByText('Deliver to Tradly Team, 75119')
    const street = screen.getByText('Kualalumpur, Malaysia')

    expect(heading).toBeOnTheScreen()
    expect(street).toBeOnTheScreen()
  })

  it('Component button must fire an action when press', () => {
    const button = screen.getByText(/change/)

    fireEvent.press(button)
    expect(mockOnPress).toBeCalledTimes(1)
  })
})
