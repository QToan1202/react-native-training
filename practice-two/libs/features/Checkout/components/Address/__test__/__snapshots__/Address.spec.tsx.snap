// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Address component Render component successfully 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "columnGap": 4,
        "flexDirection": "row",
      },
      {
        "justifyContent": "space-between",
      },
      {
        "backgroundColor": "#fff",
        "paddingHorizontal": 20,
        "paddingVertical": 15,
      },
      undefined,
    ]
  }
>
  <View
    style={
      {
        "rowGap": 4,
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "#fff",
            "fontFamily": "Montserrat_400Regular",
          },
          {
            "fontSize": 14,
          },
          {
            "color": "#4f4f4f",
            "fontFamily": "Montserrat_500Medium",
          },
        ]
      }
    >
      Deliver to Tradly Team, 75119
    </Text>
    <Text
      style={
        [
          {
            "color": "#fff",
            "fontFamily": "Montserrat_400Regular",
          },
          {
            "fontSize": 14,
          },
          [
            {
              "color": "#4f4f4f",
              "fontFamily": "Montserrat_500Medium",
            },
            {
              "opacity": 0.7,
            },
          ],
        ]
      }
    >
      Kualalumpur, Malaysia
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#33907C",
        "borderColor": "#33907C",
        "borderRadius": 50,
        "borderWidth": 2,
        "opacity": 1,
        "paddingHorizontal": 23,
        "paddingVertical": 4,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "columnGap": 6,
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          [
            {
              "color": "#fff",
              "fontFamily": "Montserrat_400Regular",
            },
            {
              "fontSize": 16,
            },
            [
              {
                "color": "#fff",
              },
              {
                "fontFamily": "Montserrat_500Medium",
                "textAlign": "center",
                "textTransform": "capitalize",
              },
              {
                "fontSize": 12,
              },
            ],
          ]
        }
      >
        change
      </Text>
    </View>
  </View>
</View>
`;
